### ISSUE TITLE
Make all multi-character flags accept `--`

### ISSUE THREAD

**Issue by bovine3dom** [tridactyl/tridactyl#4270]
We have e.g. `:winopen -private` for compatibility with Vimperator/Pentadactyl, but almost everything else in Tridactyl and every other application has `--` for long flags. We should at least make sure that `--` is accepted everywhere and is the main thing documented in help.

**Comment by GHolk** [comment]
I am tuning the search mode and the find/findnext commands.
Maybe we can use some arguments parsing library like minimist? Which can eliminate many arguments parsing code.
However, I am not sure how to import or use modules in typescript.

**Comment by bovine3dom** [comment]
we talked about that in more detail in https://github.com/tridactyl/tridactyl/issues/789

I think the problem is we have weird requirements, like not rejecting flags that aren't valid but instead treating them as normal arguments. Maybe we can use an external library for the simplest commands though?

**Comment by GHolk** [comment]
Yes, as our special requirements, I notice that most unix style arguments parser libraries in js can not stop on a invalid option.

Although @ELLIOTTCABLE 's #1863 is not finished yet, maybe I can still make use of its parser? As its document, it support many condition.

The worst condition is that I have to fork a parser library and hack some features.

**Comment by bovine3dom** [comment]
I'm sure he'd be really excited to see someone pick it up. He wrote this about the project a year or two ago https://tridactyl.xyz/ideas/#integrate-a-new-parser-into-tridactyl - I know he is really busy at the moment so I'm not sure how much guidance he could provide

**Comment by GHolk** [comment]
Bad news, the npm package he published (v1.0.0) does not work, and I can not fix it.
